lib/ftpoison/base.ex

```diff
@@ -37,33 +37,58 @@ defmodule FTPoison.Base do
       end
 
       @doc "Returns a list of remote files matching the specified path (supporting globs)"
-      @spec list(PID, String.t) :: String.t
+      @spec list(PID, String.t()) :: String.t()
       def list(pid, path) do
         case :ftp.nlist(pid, to_char_list(path)) do
           {:ok, dir_listing} -> to_string(dir_listing) |> String.split("\r\n", trim: true)
           e -> handle_error(e)
         end
       end
 
-      @spec recv(PID, String.t) :: {:ok, PID}
+      @spec recv(PID, String.t()) :: {:ok, PID}
       def recv(pid, remote_file) do
         case :ftp.recv(pid, to_char_list(remote_file)) do
           :ok -> pid
           e -> handle_error(e)
         end
       end
 
-      @spec recv(PID, String.t, String.t) :: {:ok, PID}
+      @spec recv(PID, String.t(), String.t()) :: {:ok, PID}
       def recv(pid, remote_file, local_file) do
         case :ftp.recv(pid, to_char_list(remote_file), to_char_list(local_file)) do
           :ok -> pid
           e -> handle_error(e)
         end
       end
 
-      @spec start(String.t) :: {:ok, PID} | {:error, {atom, atom}}
+      @spec send(PID, String.t()) :: :ok | {:error, {atom, atom}}
```

Comment:
are there any tests we could bring in?  I see the original functions have some basic testing.

Reply:
There are no tests for the recv methods which are used for downloading files. I added send methods for uploading files. Not sure how we would test these without actually communicating with an FTP site. We do host an FTP site for our Vendor SFTP project but I don't know that we want tests pushing files there, plus we'd have to figure out how to store encrypted credentials in this repo.

Reply:
Also, the vendor sftp isn't ftp, it's sftp.  They have similar names, but the protocols are entirely different.